<a href="https://colab.research.google.com/github/tdhakate/CVPython/blob/main/Customer_PersonalLoan_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [28]:
import pandas as pd
import numpy as np

In [29]:
# reading the CSV file into pandas dataframe
df_loan=pd.read_csv("Bank.csv")
print(df_loan)

        ID  Age  Experience  ...  CD Account  Online  CreditCard
0        1   25           1  ...           0       0           0
1        2   45          19  ...           0       0           0
2        3   39          15  ...           0       0           0
3        4   35           9  ...           0       0           0
4        5   35           8  ...           0       0           1
...    ...  ...         ...  ...         ...     ...         ...
4995  4996   29           3  ...           0       1           0
4996  4997   30           4  ...           0       1           0
4997  4998   63          39  ...           0       0           0
4998  4999   65          40  ...           0       1           0
4999  5000   28           4  ...           0       1           1

[5000 rows x 14 columns]


Below are the 14 columns(attributes) with 5000 records having index values 0 - 4999

In [30]:
df_loan.shape

(5000, 14)

In [31]:
df_loan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 14 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   ID                  5000 non-null   int64  
 1   Age                 5000 non-null   int64  
 2   Experience          5000 non-null   int64  
 3   Income              5000 non-null   int64  
 4   ZIP Code            5000 non-null   int64  
 5   Family              5000 non-null   int64  
 6   CCAvg               5000 non-null   float64
 7   Education           5000 non-null   int64  
 8   Mortgage            5000 non-null   int64  
 9   Personal Loan       5000 non-null   int64  
 10  Securities Account  5000 non-null   int64  
 11  CD Account          5000 non-null   int64  
 12  Online              5000 non-null   int64  
 13  CreditCard          5000 non-null   int64  
dtypes: float64(1), int64(13)
memory usage: 547.0 KB


In [32]:
#Below are the columns with data types
df_loan.dtypes

ID                      int64
Age                     int64
Experience              int64
Income                  int64
ZIP Code                int64
Family                  int64
CCAvg                 float64
Education               int64
Mortgage                int64
Personal Loan           int64
Securities Account      int64
CD Account              int64
Online                  int64
CreditCard              int64
dtype: object

Below are the columns with the data types

In [33]:
df_loan.describe()

ID          Age  ...       Online   CreditCard
count  5000.000000  5000.000000  ...  5000.000000  5000.000000
mean   2500.500000    45.338400  ...     0.596800     0.294000
std    1443.520003    11.463166  ...     0.490589     0.455637
min       1.000000    23.000000  ...     0.000000     0.000000
25%    1250.750000    35.000000  ...     0.000000     0.000000
50%    2500.500000    45.000000  ...     1.000000     0.000000
75%    3750.250000    55.000000  ...     1.000000     1.000000
max    5000.000000    67.000000  ...     1.000000     1.000000

[8 rows x 14 columns]

From the above description we can see 'Experience' is  '-3'. The experience cannot be a negative value and it is some error with the data.

In [34]:
df_loan.describe().transpose()

count          mean  ...       75%      max
ID                  5000.0   2500.500000  ...   3750.25   5000.0
Age                 5000.0     45.338400  ...     55.00     67.0
Experience          5000.0     20.104600  ...     30.00     43.0
Income              5000.0     73.774200  ...     98.00    224.0
ZIP Code            5000.0  93152.503000  ...  94608.00  96651.0
Family              5000.0      2.396400  ...      3.00      4.0
CCAvg               5000.0      1.937938  ...      2.50     10.0
Education           5000.0      1.881000  ...      3.00      3.0
Mortgage            5000.0     56.498800  ...    101.00    635.0
Personal Loan       5000.0      0.096000  ...      0.00      1.0
Securities Account  5000.0      0.104400  ...      0.00      1.0
CD Account          5000.0      0.060400  ...      0.00      1.0
Online              5000.0      0.596800  ...      1.00      1.0
CreditCard          5000.0      0.294000  ...      1.00      1.0

[14 rows x 8 columns]


From the dataset, the target variable is **Personal Loan**. We can predict whether a person is likely to take loan or not.

Here Family, Personal Loan, Securities Account, CD Account, Online, CreditCard are Categorical variables.

Dependent variables related to Bank are Personal Loan, Securities Account, CD( certificate of deposit)Account, Online, CreditCard.

In [36]:
# Now we count the no of records having negative experience
df_loan[df_loan['Experience'] < 0]['Experience'].count()

52

In [37]:
df_loan['Experience'] = df_loan["Experience"].abs()

In [38]:
df_loan[df_loan['Experience'] < 0]['Experience'].count()

0

Here we replace negative values with 0

In [39]:
df_loan.describe().transpose()

count          mean  ...       75%      max
ID                  5000.0   2500.500000  ...   3750.25   5000.0
Age                 5000.0     45.338400  ...     55.00     67.0
Experience          5000.0     20.134600  ...     30.00     43.0
Income              5000.0     73.774200  ...     98.00    224.0
ZIP Code            5000.0  93152.503000  ...  94608.00  96651.0
Family              5000.0      2.396400  ...      3.00      4.0
CCAvg               5000.0      1.937938  ...      2.50     10.0
Education           5000.0      1.881000  ...      3.00      3.0
Mortgage            5000.0     56.498800  ...    101.00    635.0
Personal Loan       5000.0      0.096000  ...      0.00      1.0
Securities Account  5000.0      0.104400  ...      0.00      1.0
CD Account          5000.0      0.060400  ...      0.00      1.0
Online              5000.0      0.596800  ...      1.00      1.0
CreditCard          5000.0      0.294000  ...      1.00      1.0

[14 rows x 8 columns]

In [15]:
df_loan.groupby(["Personal Loan"]).count()

ID   Age  Experience  ...  CD Account  Online  CreditCard
Personal Loan                          ...                                
0              4520  4520        4520  ...        4520    4520        4520
1               480   480         480  ...         480     480         480

[2 rows x 13 columns]

In [18]:

df_loan1= df_loan.groupby('Personal Loan')
df_loan1 = df_loan1.get_group(1)
df_loan1.shape

(480, 14)

In [19]:
df_loan1.describe().transpose()

count          mean  ...         75%      max
ID                  480.0   2390.650000  ...   3566.0000   4981.0
Age                 480.0     45.066667  ...     55.0000     65.0
Experience          480.0     19.843750  ...     30.0000     41.0
Income              480.0    144.745833  ...    172.0000    203.0
ZIP Code            480.0  93153.202083  ...  94705.5000  96008.0
Family              480.0      2.612500  ...      4.0000      4.0
CCAvg               480.0      3.905354  ...      5.3475     10.0
Education           480.0      2.233333  ...      3.0000      3.0
Mortgage            480.0    100.845833  ...    192.5000    617.0
Personal Loan       480.0      1.000000  ...      1.0000      1.0
Securities Account  480.0      0.125000  ...      0.0000      1.0
CD Account          480.0      0.291667  ...      1.0000      1.0
Online              480.0      0.606250  ...      1.0000      1.0
CreditCard          480.0      0.297917  ...      1.0000      1.0

[14 rows x 8 columns]

In [21]:
from sklearn.model_selection import train_test_split # # Import Sklearn package's data splitting function
from sklearn import metrics

X = df_loan.drop(['ID','Personal Loan'], axis=1)
y = df_loan['Personal Loan']

In [22]:
# Split X and y into training and test set in 70:30 ratio

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=10)

Logistic Regression Model:

In [23]:
from sklearn.linear_model import LogisticRegression # Import Logistic Regression machine learning library
log_model1 = LogisticRegression()

In [25]:
# Apply training data to the logistic regression model
log_model1.fit(X_train, y_train)

#Predict the test results
y_predict = log_model1.predict(X_test)

In [26]:
# Calculate the model score and print confusion matrix
log_model1_score = log_model1.score(X_test, y_test)
print("The Logistic Regression model accuracy is {}".format(log_model1_score))

The Logistic Regression model accuracy is 0.9013333333333333


In [27]:
print("The confustion matrix for Logistic Regression is")
print(metrics.confusion_matrix(y_test, y_predict))

The confustion matrix for Logistic Regression is
[[1305   55]
 [  93   47]]


The result of above confusion matrix is
True negative is 1307 & True postive is 47. False postive is 53 & False negative is 93.